Restore selection on adding formatting marks

Hornwitser opened the pull request:
Wait until after the DOM has been updated before restoring the selection in the editor. Also prevents restoring of selection when something else has been selected.

Fixes #1

feathrs commented:
This is going to really frustrate you, but I actually had made changes to the code many weeks back and then never pushed these changes (for reasons of Stupid.)

The bug still exists, but your fix will be outdated, because I added a feature(*) to show live formatting previews. I've pushed my changes, I honestly didn't figure anyone would try to contribute.

Hornwitser committed:
Refactor restore of the Editor text selection

Store the text selection to restore after formatting in an instance
variable and then prefer that over the browser selection when restoring
the selection in privew.

This avoids the need to use custom logit to sync up the textRaw, or to
save it in the DOM element.

Hornwitser commented:
> This is going to really frustrate you, but I actually had made changes to the code many weeks back and then never pushed these changes (for reasons of Stupid.)

Haha, I expected this. I fixed the bug again.
